engine/src/main/java/org/hibernate/search/analyzer/impl/LuceneAnalyzerReference.java

```diff
@@ -33,7 +33,7 @@ public void close() {
 
 	@Override
 	public <T extends AnalyzerReference> boolean is(Class<T> analyzerType) {
-		return LuceneAnalyzerReference.class.isAssignableFrom( analyzerType );
+		return analyzerType.isAssignableFrom( LuceneAnalyzerReference.class );
```

Comment:
That's what I would expect, yes: if a subclass wants to expose more of its implemented interfaces, it should do it explicitly.

That's the point of this whole `is()/`/`unwrap` pattern in this case: we don't use adaptors, so simple `instanceof` and client-side casts would have done the trick. The only reason to implement that pattern is to be a bit cleaner and to be more strict about encapsulation. See `org.hibernate.jpa.spi.AbstractEntityManagerImpl.unwrap(Class<T>)` for instance, it's even more strict than what I propose (it only accepts interface types).

Anyway... that's what I think, but it's more nitpicking than anything else, really. I don't think it'll make a difference. I'll use `getClass`, so we'll be safe if someone adds a subclass.